Rebase on message-transport

glyn opened the pull request.

glyn committed:
Rebase on message-transport

Also delete unnecessary select statements.

glyn committed:
Delete old dependency manager

ericbottard commented on glide.yaml:
```diff
@@ -0,0 +1,31 @@
+- package: github.com/giantswarm/retry-go
+  version: ~0.5.0
+- package: github.com/golang/protobuf
+  version: ~1.0.0
```
Is this really a direct dependency, or only though gRPC? If so, should we let grpc drive the version? (assuming it uses a dependency manager that glide understands)

ericbottard commented on glide.yaml:
```diff
@@ -0,0 +1,31 @@
+  subpackages:
+  - proto
+- package: github.com/projectriff/message-transport
+  subpackages:
```
Should make `version: master` explicit here, for consistency with other projects

ericbottard commented on cmd/function-sidecar.go:
```diff
@@ -115,69 +112,12 @@ func main() {
-			}
-		}
-	}()
+	carrier.Run(consumer, producer, dispatcher, output)
```
Do you think you can refactor the test in `function-sidecar_test.go` to use this and collaborators?
I really don't like the fact that it runs a compiled binary (that must exist) -- this will hinder the switch to the new Docker scheme, for one thing.

Or we could do this as a separate PR, of course

glyn replied:
@ericbottard I think that's worth trying, but in a separate PR. I raised [issue 45](https://github.com/projectriff/function-sidecar/issues/45) to cover this.

glyn committed:
Manage dependencies using glide

Direct dependencies are versioned, except for message-transport which has not
yet been released.

The protobuf dependency is versioned as it is a transitive dependency of grpc
which does not tie down the version of protobuf.

glyn commented:
@ericbottard I've force pushed a new glide dependencies commit with the changes we agreed.

ericbottard commented:
Merged as c0893eb
